code/script.js

```diff
@@ -224,102 +249,135 @@ const generateBoard = () => {
+      filterCharacters(true)
+    } else {
+      filterCharacters(false)}
+  } else if (category === 'accessories' || category === 'other'){
+      if (secretPerson.accessories.includes(checkProp) || secretPerson.other.includes(checkProp)){
+       filterCharacters(true)
+      } else {
+        filterCharacters(false)}
+    }
 }
 
-// It'll filter the characters array and redraw the game board.
+
+//...if secretPerson is matched keep is true, else keep is false. 
 const filterCharacters = (keep) => {
-  const { category, value } = currentQuestion
-  // Show the correct alert message for different categories
+const {checkProp, category, value} = currentQuestion
+
+//Different messages is passed to user based on category, than characters are filtered to keep all matches or delete all non-matches
   if (category === 'accessories') {
-    if (keep) {
-      alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
-      )
+    if (keep=== true) {
+      alert(`Yes, the person wears ${checkProp}! Let's see who are left...`)
+      charactersInPlay = charactersInPlay.filter(person => person['accessories'].includes(checkProp))
     } else {
-      alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
-      )
+      alert(`No, the person doesn't wear ${checkProp}! Let's see who are left...`)
+      charactersInPlay = charactersInPlay.filter(person => !person['accessories'].includes(checkProp))
     }
   } else if (category === 'other') {
-    // Similar to the one above
-  } else {
-    if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+    if (keep === true) {
+      alert(`Yes the person has ${checkProp}`)
+      charactersInPlay = charactersInPlay.filter(otherCheck => otherCheck.other.includes(checkProp))
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(`No, the person doesn't have ${checkProp}`)
+      charactersInPlay = charactersInPlay.filter(otherCheck => !otherCheck.other.includes(checkProp))
     }
-  }
-
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
-}
+  } else if (category === "hair"){
+    if (keep === true) {
+      alert(`Yes, the person has ${value} hair. Let's see who are left...`)
```

Comment:
Could you have written this ${value} ${category} to be able to use the same condition for both hair and eyes?
Not necessary but it would have given you less code! :)